core/trino-main/src/main/java/io/trino/operator/DirectExchangeClient.java

```diff
@@ -217,12 +223,8 @@ public Slice pollPage()
             return null;
         }
 
-        synchronized (this) {
-            if (!closed) {
-                memoryContext.setBytes(buffer.getRetainedSizeInBytes());
-                scheduleRequestIfNecessary();
-            }
-        }
+        updateRetainedMemory();
```

Comment:
is that ok to call it if `closed == true`?

Reply:
> is that ok to call it if closed == true?

It looks like it is